src/api-binder/report.ts

```diff
@@ -135,14 +134,14 @@ export async function startReporting() {
 		'deviceApiKey',
 		'appUpdatePollInterval',
 	])) as StateReportOpts;
-	// Throttle reportCurrentState so we don't query device or hit API excessively
-	const throttledReport = _.throttle(
-		reportCurrentState,
-		constants.maxReportFrequency,
-	);
+
+	let reportPending = true;
 	const doReport = async () => {
+		// Ensure no other report starts
 		if (!reportPending) {
-			await throttledReport(reportConfigs);
+			reportPending = true;
+			await reportCurrentState(reportConfigs);
```

Comment:
Oh, actually I understood what you meant. We are missing some `setTimeout` to make sure the report is triggered at the end of the throttle interval if there are calls in between.

Reply:
I've added a defer functionality so that when a request to report is throttled, the Supervisor will remember to try and report again at the next allowed (unthrottled) moment.

```
time_since_report = now - last_report_time (5000 - 2000)
defer_for = desired_frequency - time_since_report (10000 - 1500)
// delay for 8500
delay(defer_for)
try_and_report()
```

Reply:
It'll defer in the case that a report has finished and we're in the throttling period but it won't defer in the case that there's an active report in progress, eg if it's already calculated the state but is waiting on the PATCH response then it won't schedule one to report the updated state